utils/models_to_deprecate.py

```diff
@@ -0,0 +1,163 @@
+            commit_obj = repo.commit(commit_hash)
+            committed_datetime = commit_obj.committed_datetime
+            models_info[model]["commit_hash"] = commit_hash
+            models_info[model]["first_commit_datetime"] = committed_datetime
+            models_info[model]["model_path"] = model_path
+
+        # Filter out models which were added less than a year ago
+        models_info = {
+            model: info for model, info in models_info.items() if info["first_commit_datetime"] < thresh_date
+        }
+
+        # For each model on the hub, find it corresponding model based on its tags and update the number of downloads
+        for model, info in models_info.items():
+            models_info[model]["downloads"] = 0
+
+        for i, hub_model in enumerate(api.list_models()):
+            if i % 100 == 0:
+                print(f"Processing model {i}")
+            if max_num_models != -1 and i > max_num_models:
+                break
+            if hub_model.private:
+                continue
+            for tag in hub_model.tags:
+                tag = tag.lower().replace("-", "_")
+                if tag in models_info:
+                    models_info[tag]["downloads"] += hub_model.downloads
+
+    if save_model_info:
+        # Make datetimes serializable
+        for model, info in models_info.items():
+            info["first_commit_datetime"] = info["first_commit_datetime"].isoformat()
+        with open("models_info.json", "w") as f:
+            json.dump(models_info, f, indent=4)
+
+    print("\nModels to deprecate:")
+    n_models_to_deprecate = 0
+    models_to_deprecate = {}
+    for model, info in models_info.items():
+        n_downloads = info["downloads"]
+        if n_downloads < thresh_num_downloads:
+            n_models_to_deprecate += 1
+            models_to_deprecate[model] = info
+            print(f"\nModel: {model}")
+            print(f"Downloads: {n_downloads}")
+            print(f"Date: {info['first_commit_datetime']}")
+    print(f"\nNumber of models to deprecate: {n_models_to_deprecate}")
+    print("Before deprecating make sure to verify the models, including if they're used as a module in other models.")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--save_model_info", action="store_true", help="Save the retrieved model info to a json file.")
+    parser.add_argument(
+        "--use_cache", action="store_true", help="Use the cached model info instead of calling the hub."
+    )
+    parser.add_argument(
+        "--thresh_num_downloads",
+        type=int,
+        default=5_000,
+        help="Threshold number of downloads below which a model should be deprecated. Default is 1,000.",
+    )
+    parser.add_argument(
+        "--thresh_date",
+        type=str,
+        default=None,
+        help="Date to consider the first commit from. Format: YYYY-MM-DD, default is one year ago.",
```

Comment:
> If not specified, the date of one year ago from the current date will be used.

Since the default is `None`, maybe a bit of rephrase is better ? Like above (or whatever you find better)

Reply:
I've updated in [98183b9](https://github.com/huggingface/transformers/pull/29686/commits/98183b9901e0828be6763dd280d45fbb473aba39). LMKWYT